capped_file.go

```diff
@@ -0,0 +1,195 @@
+// bagdb: Simple datastorage
+// Copyright 2021 billy authors
+// SPDX-License-Identifier: BSD-3-Clause
+
+package billy
+
+import (
+	"fmt"
+	"os"
+)
+
+// cappedFile has an API-surface as if it were one file, but maps
+// to a set of files on disk. These files are all capped in size
+// to maxFileSize.
+type cappedFile struct {
+	cap   uint64
+	files []*os.File
+}
+
+// newCappedFile creates a cappedFile
+func newCappedFile(basename string, nFiles int, cap uint64, readonly bool) (*cappedFile, error) {
+	flags := os.O_RDWR | os.O_CREATE
+	if readonly {
+		flags = os.O_RDONLY
+	}
+	var files []*os.File
+	for i := 0; i < nFiles; i++ {
+		var (
+			f   *os.File
+			err error
+		)
+		if i == 0 {
+			f, err = os.OpenFile(basename, flags, 0666)
+		} else {
+			f, err = os.OpenFile(fmt.Sprintf("%v.cap.%d", basename, i), flags, 0666)
+		}
+		if err != nil {
+			// Clean-up: close opened files
+			for _, f := range files {
+				f.Close()
+			}
+			return nil, err
+		}
+		files = append(files, f)
+	}
+	return &cappedFile{
+		cap:   cap,
+		files: files,
+	}, nil
+}
+
+// Write implements io.Writer
+func (cf *cappedFile) Write(p []byte) (n int, err error) {
+	return cf.WriteAt(p, 0)
```

Comment:
Are you sure this is correct? AFAIK a file should have it's own internal pointer as to where it is at currently and subsequent calls to write should keep moving that forward (also, positioned at the end initially?) vs. just keeping overwriting offset 0.

Reply:
I think we're not really using read/write at all, might as well drop it and get rid of this issue?

Reply:
I agree with adding an offset. Would love to get rid of it, but doesn't seem doable